ztp/gitops-subscriptions/argocd/PowerConfiguration.md

```diff
@@ -0,0 +1,224 @@
+
+The tuned performance-patch can be used to confine the maximum cpu frequency. The profile section of the performance-patch can be extended to use the `sysfs` plugin to set the `max_perf_pct`, which applies to all cpus. The `max_perf_pct` parameter controls the maximum frequency the cpufreq driver is allowed to set as a percent of the maximum supported cpu frequency (/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo\_max\_freq). This is demonstrated in the example below.
+
+> **Note** `/sys/devices/system/cpu/intel_pstate/max_perf_pct=x` where `x` is to be configured based on the hardware specification and powersaving requirements.
+
+```yaml
+  sourceFiles:
+    - fileName: PerformanceProfile.yaml
+      policyName: "config-policy"
+      metadata:
+        name: openshift-node-performance-profile
+      spec:
+        additionalKernelArgs:
+          - "cpufreq.default_governor=schedutil"
+          - "rcupdate.rcu_normal_after_boot=0"
+          - "efi=runtime"
+        ...
+        workloadHints:
+          highPowerConsumption: false
+          perPodPowerManagement: true
+        ...
+    - fileName: TunedPerformancePatch.yaml
+      policyName: "config-policy"
+      spec:
+        profile:
+          - name: performance-patch
+            data: |
+              ...
+              [sysfs]
+              /sys/devices/system/cpu/intel_pstate/max_perf_pct=72
+```
+
+### 4) [Mix of Power saving and Performance mode](#mix-power-performance-mode)
+
+The SNO can be deployed to a default Power saving mode as shown above. For critical workloads that require the highest performance and lowest latency, each such pod should be annotated to disable c-states and set the cpufreq governor to `performance`.
+
+The pod spec should be configured as follows:
+
+```yaml
+metadata:
+  annotations:
+    cpu-c-states.crio.io: "disable"
+    cpu-freq-governor.crio.io: "performance"
+```
+
+It is worthwhile to note the following:
+
+- These annotations are likely to be used with the other high performance cri-o annotation described [here](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/scalability_and_performance/cnf-low-latency-tuning#doc-wrapper).
+- These annotations have the same restrictions as the other high performance cri-o annotations:
+    - The pod must use the performance-&lt;profile-name&gt; runtime class.
+    - The pod must have a [QoS Class of guaranteed](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed) (i.e. uses whole CPUs).
+
+## Migrating from one mode to another
+
+This section outlines the steps and configurations for migrating from one mode to another mode.
+
+### 1) Performance to Power saving
```

Comment:
I think this section should mention that the crio annotations need to be added to any critical workload pods before this migration is done.